frontend/src/components/pages/VerifyEmail.tsx

```diff
@@ -1,21 +1,78 @@
-import React, { useContext, useEffect } from "react";
-import { Flex, Image, VStack, Center, Text } from "@chakra-ui/react";
-import { useQuery } from "@apollo/client";
+import React, {
+  Dispatch,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { Flex, Image, VStack, Center, Text, Button } from "@chakra-ui/react";
+import { useQuery, useMutation } from "@apollo/client";
 import { Redirect } from "react-router-dom";
 import RHSLogo from "../../assets/RHSlogo.png";
 import AuthContext from "../../contexts/AuthContext";
 import { GET_EMAIL_VERIFIED_BY_EMAIL } from "../../APIClients/queries/UserQueries";
 import { MANAGE_COURSES_PAGE } from "../../constants/Routes";
 import authAPIClient from "../../APIClients/AuthAPIClient";
+import { SEND_EMAIL_VERIFICATION_LINK } from "../../APIClients/mutations/AuthMutations";
+
+interface VerifyEmailProps {
+  hasError: boolean;
+  hasBeenPressed: boolean;
+  setHasBeenPressed: Dispatch<SetStateAction<boolean>>;
+}
+
+const VerifyEmailContent = ({
```

Comment:
Wasn't sure where I should put this component so I left it in the same file, feel free to let me know where I should put this.

Reply:
Seems fine to me to keep it here, this is a short file anyways.

Reply:
Yeah, more like I know by principle, some people like having one component per file.